src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -679,7 +698,32 @@ Leader::UpdateStatus Leader::UpdateTlv(NetworkDataTlv &aTlv, const NetworkDataTl
     return status;
 }
 
-void Leader::RegisterNetworkData(uint16_t aRloc16, const NetworkData &aNetworkData)
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_SIGNAL_NETWORK_DATA_FULL
+Error Leader::CanRegisterNetworkData(const NetworkData &aNetworkData, uint16_t aOldRloc16) const
+{
+    // Create a clone of the leader's network data, and try to register
+    // `aNetworkData` into the copy (as if this device itself is the
+    // leader). `mIsClone` flag is used to mark the clone and ensure
+    // that the cloned instance does interact with other OT modules,
+    // e.g., does not start timer, or does not signal version change
+    // using `Get<ot::Notifier>().Signal()`, or allocate service or
+    // context ID.
+
+    Leader leaderClone(GetInstance());
+
+    leaderClone.mIsClone = true;
```

Comment:
I'm not sure using a class level `mIsClone` is the best, it requires that all member functions of Leader class aware that the leader object can be a clone, which can be error prone when some one add a new function but forget to check `mIsClone`

I wonder if it is possible to convert this member field into a boolean argument, and propagate this argument to all functions need to be aware this behavior. Because the lifespan of the cloned leader is less than the current function scope, the argument should be sufficient to pass this information.

Reply:
> I wonder if it is possible to convert this member field into a boolean argument, and propagate this argument to all functions need to be aware this behavior.

This can add overhead of passing the arg around (to a lot of methods, almost all of them). Also I feel it has the same issue that when writing new code we should remember to pass this to new methods and make sure to handle it (would be similar issue as forgetting to check `mIsClone`).